Tests/AsyncChannelsTests/AsyncChannelTests.swift

```diff
@@ -266,6 +266,112 @@ final class AsyncTest: XCTestCase {
         let r = await result.reduce(into: []) { $0.append($1) }
         XCTAssertEqual(["foo", "bar"].sorted(), r.sorted())
     }
+    
+    func testOptionalSomeSelect() async {
+        let a = Channel<String>()
+        let result = Channel<String>(capacity: 1)
+        
+        Task {
+            await a <- "foo"
+        }
+        
+        await select {
+            if true {
+                receive(a) { await result <- $0! }
+            }
+        }
+
+        result.close()
+        
+        let r = await result.reduce(into: []) { $0.append($1) }
+        XCTAssertEqual(["foo"], r)
+    }
+    
+    func testOptionalNoneSelect() async throws {
```

Comment:
Perhapse a better way to write this test would be to use a `none` case in the select ex:

```swift
func testOptionalNoneSelect() async throws {
    let a = Channel<String>()
    let done = Channel<Bool>()
    
    Task {
        await a <- "foo"
    }
    
    Task {
        await select {
            if false {
                receive(a) { XCTFail() }
            }
            none {
                await done <- true
            }
        }
    }

    await <-done
    a.close()
}
```

This avoids the need for the yield loop and simplifies the logic slightly.

Reply:
I agree that using `none` is better as you suggested. I fixed it at [f9b1793](https://github.com/gh123man/Async-Channels/pull/10/commits/f9b179350d62c992cb423e1df9d6fa0b1b10a254).